deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts

```diff
@@ -101,54 +97,87 @@ export class MigrationAssistanceStack extends Stack {
                 }
             }
         });
-        this.mskARN = mskCluster.attrArn
-
-        const comparatorSQLiteSG = new SecurityGroup(this, 'comparatorSQLiteSG', {
-            vpc: props.vpc,
-            allowAllOutbound: false,
+        new StringParameter(this, 'SSMParameterMSKARN', {
+            description: 'OpenSearch Migration Parameter for MSK ARN',
+            parameterName: `/migration/${props.stage}/${props.defaultDeployId}/mskClusterARN`,
+            stringValue: mskCluster.attrArn
         });
-        comparatorSQLiteSG.addIngressRule(comparatorSQLiteSG, Port.allTraffic());
-
-        // Create an EFS file system for the traffic-comparator
-        const comparatorSQLiteEFS = new FileSystem(this, 'comparatorSQLiteEFS', {
-            vpc: props.vpc,
-            securityGroup: comparatorSQLiteSG
+        new StringParameter(this, 'SSMParameterMSKClusterName', {
+            description: 'OpenSearch Migration Parameter for MSK cluster name',
+            parameterName: `/migration/${props.stage}/${props.defaultDeployId}/mskClusterName`,
+            stringValue: mskCluster.clusterName
         });
 
+        if (props.trafficComparatorEnabled) {
```

Comment:
My main reason for placing here is to make things cleaner across the services to act more independently. For instance the Replayer output security group is needed by the console and the replayer. If we ever have a situation where we want to deploy a console without a replayer we would hit an error or the console would have to add logic to only add certain settings if the replayer is enabled.

Reply:
> If we ever have a situation where we want to deploy a console without a replayer we would hit an error or the console would have to add logic to only add certain settings if the replayer is enabled.

If a user was deploying the console without a replayer, limiting the privileges of the console is probably the more secure choice instead of creating an unused security group to avoid an error. That said, I know the console currently has several such dependencies so we should create an issue to track mitigating all such coupling. Would you mind creating a backlog issue to track this?

Reply:
Have created a task to re-evaluate how we think about shared resources in CDK in these scenarios